From 1b7cac1cbdb7078f575a3222be451a9bf1ac35ec Mon Sep 17 00:00:00 2001
From: Alyssa Ross <hi@alyssa.is>
Date: Wed, 31 Jan 2024 15:33:02 +0100
Subject: [PATCH] build: add missing dependency to gdkpixbuf_dep

This should match the dependencies passed to the library() call that
creates gdkpixbuf.  Otherwise, linking the gdkpixbuf_bin executables
will fail if -Ddefault_library=static, because static libraries don't
carry dependency information themselves.
---
Link: https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/161

 gdk-pixbuf/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk-pixbuf/meson.build b/gdk-pixbuf/meson.build
index a11926eee..450484d68 100644
--- a/gdk-pixbuf/meson.build
+++ b/gdk-pixbuf/meson.build
@@ -269,7 +269,7 @@ endif
 gdkpixbuf_dep = declare_dependency(
   link_with: gdkpixbuf,
   include_directories: root_inc,
-  dependencies: gdk_pixbuf_deps,
+  dependencies: [ gdk_pixbuf_deps, included_loaders_deps ],
   sources: [ gdkpixbuf_enum_h, built_girs ],
 )
 meson.override_dependency('gdk-pixbuf-2.0', gdkpixbuf_dep)
-- 
GitLab

